[ML] Anomaly Detection datafeed chart: updates tooltip copy for clarity

alvarezmelissa87 opened the pull request:
## Summary

Related meta issue: https://github.com/elastic/kibana/issues/143330

Updates tooltip copy for clarity.

<img width="701" alt="image" src="https://user-images.githubusercontent.com/6446462/196272244-7813e6f5-a51f-4d18-a70b-0a91bcaf968a.png">




### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

alvarezmelissa87 commented:
@lcawl - do you have any thoughts/suggestions about the copy?

droberts195 commented:
I added a comment to https://github.com/elastic/kibana/issues/143330 to say that the fix should go further than just the labels, and also display both lines in the tooltip even when one of the numbers isn't available.

I guess that aspect is much harder to fix, so if you want to defer it to a followup PR that's fine.  But in that case please can you leave #143330 open to make clear that there's a second part to fixing it overall.

lcawl commented on x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_chart_flyout/datafeed_chart_flyout.tsx:
```diff
@@ -510,7 +510,7 @@ export const DatafeedChartFlyout: FC<DatafeedChartFlyoutProps> = ({
                           color={euiTheme.euiColorPrimary}
                           id={i18n.translate('xpack.ml.jobsList.datafeedChart.sourceSeriesId', {
-                            defaultMessage: 'Source indices',
+                            defaultMessage: 'Source indices document count',
```
IMO we could make these phrases more similar in layout (so the different intent is clearer). For example:
```suggestion
                            defaultMessage: 'Count of documents in source indices',
```

peteharverson commented on x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_chart_flyout/datafeed_chart_flyout.tsx:
```diff
@@ -523,7 +523,7 @@ export const DatafeedChartFlyout: FC<DatafeedChartFlyoutProps> = ({
                           color={euiTheme.euiColorAccentText}
                           id={i18n.translate('xpack.ml.jobsList.datafeedChart.bucketSeriesId', {
-                            defaultMessage: 'Job results',
+                            defaultMessage: 'Count of documents read by datafeed',
```
Looking at what we use in the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/8.4/ml-get-bucket.html#ml-get-bucket-results) for `event_count`, I tried replacing `read` by `processed` but don't like the way it wraps on to 2 lines!

<img width="280" alt="image" src="https://user-images.githubusercontent.com/7405507/195895351-69c8e71e-b66c-4c1d-8189-33ba4770049d.png">

I can't think of a better, short alternative to what you have here. Over to you @lcawl !

Is there anything you can do to stop the truncation of these labels in the legend?

<img width="514" alt="image" src="https://user-images.githubusercontent.com/7405507/195895805-26033f46-e3a8-4165-ae7e-25da4cfe3b65.png">

alvarezmelissa87 replied:
The truncation is built into the elastic charts `LineSeries` component - so a change would require us updating the style of the internal component directly 🤔 Not sure we want to do that here. Maybe we should change the tooltips so that we can still see 'Source indices' and 'Datafeed' at the start so that it's still clear even with truncation?
cc @lcawl

lcawl replied:
Yup, putting the important phrases at the front LGTM

alvarezmelissa87 commented:
@elasticmachine merge upstream

alvarezmelissa87 commented:
> I added a comment to #143330 to say that the fix should go further than just the labels, and also display both lines in the tooltip even when one of the numbers isn't available.
> 
> I guess that aspect is much harder to fix, so if you want to defer it to a followup PR that's fine. But in that case please can you leave #143330 open to make clear that there's a second part to fixing it overall.

Added an item to cover this in the issue 👍 Thank you - good suggestion for extra clarity!

peteharverson reviewed (approved):
Latest changes LGTM

lcawl reviewed (approved):
UI text changes LGTM
